www/perl-lib/SMT/Rest/SCCAPIv4.pm

```diff
@@ -348,6 +351,92 @@ sub delete_system
     return (Apache2::Const::OK, {});
 }
 
+sub delete_single_product
+{
+    my $self = shift || return (undef, undef);
+
+    my $regsharing = SMT::Utils::hasRegSharing();
+    if ( $regsharing ) {
+        return (
+            Apache2::Const::HTTP_UNPROCESSABLE_ENTITY,
+            "Single product deactivation is not available when registration sharing is enabled"
+        );
+    }
+
+    # We are sure, that user is a system GUID
+    my $guid = $self->user();
+    my $c    = JSON::decode_json($self->read_post());
+
+    #
+    # lookup the Clients target
+    #
+    my $target = SMT::Utils::lookupTargetForClient($self->dbh(), $guid, $self->request());
+
+    #
+    # find Catalogs
+    #
+    my $existingregs = SMT::Utils::lookupRegistrationByGUID($self->dbh(), $guid, $self->request());
+    my @pidarr = keys %{$existingregs};
+    my $catalogs = SMT::Registration::findCatalogs($self->request(), $self->dbh(), $target, \@pidarr);
+
+    if ( (keys %{$catalogs}) == 0)
+    {
+        return (Apache2::Const::HTTP_UNPROCESSABLE_ENTITY, "No repositories found");
+    }
+
+    foreach my $param ( qw{identifier version arch} ) {
+        unless ( exists $c->{$param} && $c->{$param} ) {
+            return (Apache2::Const::HTTP_UNPROCESSABLE_ENTITY, "Missing required parameter: $param");
+        }
+    }
+
+    unless (exists $c->{release_type}) {
+        $c->{release_type} = undef;
+    }
+
+    my ($v, $r) = split(/-/, $c->{version}, 2);
+    $c->{version} = $v;
+
+    my $productId = SMT::Utils::lookupProductIdByName($self->dbh(), $c->{identifier}, $c->{version},
+        $c->{release_type}, $c->{arch});
+    if(not $productId)
+    {
+        return (Apache2::Const::HTTP_UNPROCESSABLE_ENTITY, "No valid product found");
+    }
+
+    my @statements;
+
+    push @statements, sprintf("DELETE FROM Registration WHERE GUID=%s AND PRODUCTID=%s",
+        $self->dbh()->quote($guid),
+        $self->dbh()->quote($productId));
+
+    foreach my $key_name (
+        (
+            "product-name-$productId", "product-version-$productId", "product-arch-$productId",
+            "product-rel-$productId", "product-token-$productId",
+        )
+    ) {
+        push @statements, sprintf("DELETE FROM MachineData WHERE GUID = %s AND KEYNAME = %s",
+            $self->dbh()->quote($guid),
+            $self->dbh()->quote( $key_name ),
+        );
+    }
+
+    foreach my $statement ( @statements ) {
+        $self->request()->log->info("STATEMENT: $statement");
+        eval {
+            $self->dbh()->do($statement);
+        };
+        if($@)
+        {
+            return (Apache2::Const::SERVER_ERROR, "DBERROR: ".$self->dbh()->errstr);
+        }
+    }
+
```

Comment:
Seems that this product removal is not submitted to SCC. Is that by intention?
I would suggest to add it, otherwise this is the first bug you will get when this feature gets public.